ravenframework/Optimizers/GeneticAlgorithm.py

```diff
@@ -677,27 +1034,67 @@ def _collectOptPoint(self, rlz, fitness, objectiveVal, g):
       @ In, fitness, xr.DataArray, fitness values at each chromosome of the realization
       @ Out, point, dict, point used in this realization
     """
-
     varList = list(self.toBeSampled.keys()) + self._solutionExport.getVars('input') + self._solutionExport.getVars('output')
     varList = set(varList)
     selVars = [var for var in varList if var in rlz.data_vars]
     population = datasetToDataArray(rlz, selVars)
-    optPoints, fit, obj, gOfBest = zip(*[[x,y,z,w] for x, y, z,w in sorted(zip(np.atleast_2d(population.data),
-                                                                            np.atleast_1d(fitness.data),
-                                                                            objectiveVal,
-                                                                            np.atleast_2d(g.data)),
-                                                                        reverse=True,
-                                                                        key=lambda x: (x[1]))])
-    point = dict((var,optPoints[0][i]) for i, var in enumerate(selVars) if var in rlz.data_vars)
+    optPoints,fit,obj,gOfBest = zip(*[[x,y,z,w] for x, y, z,w in sorted(zip(np.atleast_2d(population.data),
+                                                                              datasetToDataArray(fitness, self._objectiveVar).data,
+                                                                              objectiveVal,np.atleast_2d(g.data)),
+                                                                          reverse=True,
+                                                                          key=lambda x: (x[1]))])
+    point = dict((var,float(optPoints[0][i])) for i, var in enumerate(selVars) if var in rlz.data_vars)
```

Comment:
@joshua-cogliati-inl Can you remove "float" keyword here? Using float here to prevent the use of string values in realizations. This is also the request from @Nhatkhang.

Reply:
Yes, the optimizer tests pass if it is removed, so I am removing it in abdda88f097fabea4e7f81d12b958f7edf6173ae